src/common/services/PPAPIService.ts

```diff
@@ -105,4 +108,69 @@ export class PPAPIService {
             (websitePreviewId ? `/${websitePreviewId}` : '') +
             `?api-version=${PPAPI_WEBSITES_API_VERSION}`;
     }
+
+    static async getGovernanceFlag(
+        serviceEndpointStamp: ServiceEndpointCategory,
+        environmentId: string,
+        sessionId: string,
+        copilotGovernanceSetting: string,
+        websiteId: string | null
+    ): Promise<boolean> {
+        try {
+            let governanceEndpoint: string;
+            const accessToken = await powerPlatformAPIAuthentication(serviceEndpointStamp, true);
+            const ppBaseEndpoint = await PPAPIService.getPPAPIServiceEndpoint(serviceEndpointStamp, environmentId);
+
+            // Build governance endpoint URL based on whether website ID is provided
+            if (websiteId) {
+                // When site context is present.
+                const websiteDetails = await this.getWebsiteDetailsByWebsiteRecordId(serviceEndpointStamp, environmentId, websiteId);
+                governanceEndpoint = `${ppBaseEndpoint.split('?')[0]}/${websiteDetails?.id}/governance/${copilotGovernanceSetting}?api-version=${PPAPI_WEBSITES_API_VERSION}`;
+            } else {
+                // Using copilot without site context.
+                const envEndpoint = ppBaseEndpoint.split('/websites')[0];
+                governanceEndpoint = `${envEndpoint}/governance/${copilotGovernanceSetting}?api-version=${PPAPI_WEBSITES_API_VERSION}`;
+            }
+
+            const response = await fetch(governanceEndpoint, {
+                method: 'GET',
+                headers: getCommonHeaders(accessToken)
+            });
+
+            if (response.ok) {
+                const result = await response.json();
+                const allowProDevCopilots = ["all", "include", "exclude", "true"].includes(result.toString().toLowerCase());
+
+                sendTelemetryEvent({
+                    eventName: VSCODE_EXTENSION_GOVERNANCE_CHECK_SUCCESS,
+                    environmentId: environmentId,
+                    websiteId: websiteId || '',
+                    copilotSessionId: sessionId,
+                    copilotGovernanceResponse: allowProDevCopilots
```

Comment:
Should be `result` instead of `allowProDevCopilots`, right?

Reply:
Right, updated.